arch/arm/soc/nordic_nrf5/nrf52/power.c

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2017 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <soc_power.h>
+#include <nrf_power.h>
+
+/* System_OFF is deepest Power state available, On exiting from this
+ * state CPU including all peripherals reset
+ */
+static void _system_off(void)
+{
+	nrf_power_system_off();
+}
+
+void configure_ram_retention(void)
+{
+	/* Place Holder for RAM retention */
+}
+
+void _issue_low_power_command(void)
+{
+	__SEV();
+	__WFE();
+	__WFE();
+	if (NRF_GPIOTE->EVENTS_PORT) {
+		NRF_GPIOTE->EVENTS_PORT = 0;
+	}
+}
+
+static void _low_power_mode(enum power_states state)
+{
+	switch (state) {
+	case SYS_POWER_STATE_CPU_LPS:
+		nrf_power_task_trigger(NRF_POWER_TASK_CONSTLAT);
+		break;
+
+	case SYS_POWER_STATE_CPU_LPS_1:
+		nrf_power_task_trigger(NRF_POWER_TASK_LOWPWR);
+		break;
+
+	default:
+		/* Unsupported State */
+		break;
+	}
+
+	/* Wait for __WFE*/
+	_issue_low_power_command();
```

Comment:
I just read in OPS 20.2 Port event:
> Putting the system into System ON IDLE while DETECT is high will not cause DETECT to wake the system
up again.

@anang I;m not sure whether we need a check/prevent this just  before lpw cmd is issued?

Reply:
Yes I got your point, But I am not getting any reference, how we can see if DETECT is high or low. 
My understanding is - 
Detect will remain High for the duration till GPIO is Low (incase configured Low as in our Case). https://devzone.nordicsemi.com/question/171620/clear-gpio-sense-signal-on-hitolo/

So i think we need to check our application if GPIO is high and allow attempting Low Power state if GPIO is high. What is your suggestion on this ?

Reply:
Guys looks like our merge window is approaching, Can you please provide comments on this. 

Basic purpose of this patch is to have basic skeleton for nrf Low power state and Deep sleep state. If any enhancement (like restore context on wake up) is to be done we can take that up later ?

Reply:
@youvedeep 
> So i think we need to check our application if GPIO is high and allow attempting Low Power state if GPIO is high.

OK - we can improve this in future PR if we will want.

Reply:
But it will be great if add a remark(comment) near the sleep call execution.
When you will add assertions i requested this patch will be ready for me.

Reply:
Thanks @nvlsianpu  I will add assertion :), 

Also good idea, 
We will check GPIO status before attempting next sleep entry so that will make sure DETECT is Low.

Reply:
Also I was thinking of one more change. 

Currently I have am calling "configure_ram_retention(void)" inside SOC specific file. But i feel it should be left to application developer which RAM blocks she want to configure based on his application. So I am thinking of moving that to Application, just before entertaining into Low Power states.

Reply:
Agree on that - the setup of Backups-regs, Ram retention etc. should be a separate operation.